File: platform/packages_py/auth_config/auth_config/get_required_credentials.py
from typing import List
from .constants.credential_requirements import CREDENTIAL_REQUIREMENTS
from .logger import create_logger

logger = create_logger('fetch_auth_config', __file__)

def get_required_credentials(auth_type: str) -> List[str]:
    """
    Look up required credentials for an auth type.
    
    Args:
        auth_type: AuthType string
        
    Returns:
        List of required field names
    """
    reqs = CREDENTIAL_REQUIREMENTS.get(auth_type, [])
    logger.debug(f'get_required_credentials for {auth_type}: {reqs}')
    return reqs

File: platform/packages_mjs/auth_config/src/get-required-credentials.ts
import { AuthType } from '@internal/auth-encoding';
import { CREDENTIAL_REQUIREMENTS } from './constants/credential-requirements.js';
import { createLogger } from './logger.js';

const logger = createLogger('fetch_auth_config', import.meta.url);

export function getRequiredCredentials(authType: AuthType): string[] {
    const reqs = CREDENTIAL_REQUIREMENTS[authType] || [];
    logger.debug(`getRequiredCredentials for ${authType}:`, { reqs });
    return reqs;
}
